sdks/wasm/binding_support.js

```diff
@@ -693,8 +695,27 @@ var BindingSupportLib = {
 				var gc_handle = obj.__mono_gchandle__;
 				if (typeof gc_handle  !== "undefined") {
 					this.wasm_unbind_js_obj_and_free(js_id);
-					delete obj.__mono_gchandle__;
-					delete obj.__mono_jshandle__;
+
+					// Windows Edge and Windows IE throws an exception `Object doesn't support this action` on
+					// certain host objects.
+					try {
+						delete obj.__mono_gchandle__;
```

Comment:
given the need for an exception handler and that we can just do ```x.__mono_gchandle__ !== undefined``` to cover all cases, should we even bother using ```delete```? It deoptimizes code in some runtimes as well.

Reply:
So we just set `obj.__mono_gchandle__ = undefined` instead of trying to delete?

Reply:
Did not know it deoptimized code in some runtimes.  Good to know.

Reply:
Yeah there are specific primitives (the delete statement is one of them) that make runtimes unhappy because they can change object shapes and stuff. It gets nasty since key/value pairs on JS objects are ordered, so if you do ```var temp = a.x; delete a.x; a.x = temp;``` you've actually mutated the object in an observable way.

try/catch and the 'arguments' object are two other ones that sometimes cause deopts.

In this case I think writing undefined to the fields should be good enough, and it eliminates the need for that special case. delete is only (imo) necessary if you're using the object's keys themselves for some important reason (like a dictionary) instead of just the values.

Reply:
Will do and thanks for the explanation.